crates/sui-package-alt/src/sui_flavor.rs

```diff
@@ -21,16 +26,103 @@ impl MoveFlavor for SuiFlavor {
 
     type PublishedMetadata = (); // TODO
 
-    type EnvironmentID = String; // TODO
-
     type AddressInfo = (); // TODO
 
     type PackageMetadata = (); // TODO
 
     fn implicit_deps(
-        &self,
-        environments: impl Iterator<Item = Self::EnvironmentID>,
-    ) -> DependencySet<PinnedDependencyInfo> {
-        todo!()
+        file_handle: FileHandle,
+        environments: impl Iterator<Item = EnvironmentID>,
+    ) -> DependencySet<CombinedDependency> {
+        let mut deps = DependencySet::new();
+        let deps_to_skip = ["DeepBook".to_string()];
+        let packages = latest_system_packages();
+        let sha = &packages.git_revision;
+        for env in environments {
+            let pkgs = packages
+                .packages
+                .iter()
+                .filter(|package| !deps_to_skip.contains(&package.package_name));
+
+            for package in pkgs {
+                let repo = SYSTEM_GIT_REPO.to_string();
+                let dep = ManifestDependencyInfo::Git(ManifestGitDependency {
+                    repo: repo.clone(),
+                    rev: Some(sha.clone()),
+                    subdir: PathBuf::from(&package.repo_path),
+                });
+
+                let dep_info = CombinedDependency::new(Dependency::new(
+                    dep,
+                    env.clone(),
```

Comment:
perhaps it doesn't matter since the system dependencies probably won't have interesting environments, but I'm not sure we actually want to use `env` here. I'm not sure what we do want to use though.

The problem is that `env` is the depending package's environment name and could be anything, but the system packages won't have all those environments defined.

As I'm writing this I'm realizing it's more complicated, because I'm not sure what environments we should actually put in the lock file for the system packages. Probably none, but then we need to put the publication information somewhere. Hm.

Reply:
Maybe we do actually want to not have lockfiles for the system packages, and just use the `published-at` feature for them.